engine/storage/snapshot/src/org/apache/cloudstack/storage/snapshot/XenserverSnapshotStrategy.java

```diff
@@ -279,7 +284,52 @@ public boolean deleteSnapshot(Long snapshotId) {
 
     @Override
     public boolean revertSnapshot(SnapshotInfo snapshot) {
-        throw new CloudRuntimeException("revert Snapshot is not supported");
+        if (canHandle(snapshot,SnapshotOperation.REVERT) == StrategyPriority.CANT_HANDLE) {
+            throw new UnsupportedOperationException("Reverting not supported. Create a template or volume based on the snapshot instead.");
```

Comment:
``UnsupportedOperationException`` means that a method implementation is not supported on a class.  It seems that ``CloudRuntimeException`` would be a more appropriate exception to throw under these circumstances.